src/FlexVotingClient.sol

```diff
@@ -0,0 +1,209 @@
+    require(
+      _proposalVote.forVotes + _proposalVote.againstVotes + _proposalVote.abstainVotes > 0,
+      "no votes expressed"
+    );
+    uint256 _proposalSnapshotBlockNumber = GOVERNOR.proposalSnapshot(proposalId);
+
+    // We use the snapshot of total raw balances to determine the weight with
+    // which to vote. We do this for two reasons:
+    //   (1) We cannot use the proposalVote numbers alone, since some people with
+    //       balances at the snapshot might never express their preferences. If a
+    //       large holder never expressed a preference, but this contract nevertheless
+    //       cast votes to the governor with all of its weight, then other users may
+    //       effectively have *increased* their voting weight because someone else
+    //       didn't participate, which creates all kinds of bad incentives.
+    //   (2) Other people might have already expressed their preferences on this
+    //       proposal and had those preferences submitted to the governor by an
+    //       earlier call to this function. The weight of those preferences
+    //       should still be taken into consideration when determining how much
+    //       weight to vote with this time.
+    // Using the total raw balance to proportion votes in this way means that in
+    // many circumstances this function will not cast votes with all of its
+    // weight.
+    uint256 _totalRawBalanceAtSnapshot = getPastTotalBalance(_proposalSnapshotBlockNumber);
+
+    // We need 256 bits because of the multiplication we're about to do.
+    uint256 _votingWeightAtSnapshot = IVotingToken(address(GOVERNOR.token())).getPastVotes(
+      address(this), _proposalSnapshotBlockNumber
+    );
+
+    //      forVotesRaw          forVoteWeight
+    // --------------------- = ------------------
+    //     totalRawBalance      totalVoteWeight
+    //
+    // forVoteWeight = forVotesRaw * totalVoteWeight / totalRawBalance
+    uint128 _forVotesToCast = SafeCast.toUint128(
+      (_votingWeightAtSnapshot * _proposalVote.forVotes) / _totalRawBalanceAtSnapshot
+    );
+    uint128 _againstVotesToCast = SafeCast.toUint128(
+      (_votingWeightAtSnapshot * _proposalVote.againstVotes) / _totalRawBalanceAtSnapshot
+    );
+    uint128 _abstainVotesToCast = SafeCast.toUint128(
+      (_votingWeightAtSnapshot * _proposalVote.abstainVotes) / _totalRawBalanceAtSnapshot
+    );
+
+    // This param is ignored by the governor when voting with fractional
+    // weights. It makes no difference what vote type this is.
+    uint8 unusedSupportParam = uint8(VoteType.Abstain);
+
+    // Clear the stored votes so that we don't double-cast them.
+    delete proposalVotes[proposalId];
+
+    bytes memory fractionalizedVotes =
+      abi.encodePacked(_againstVotesToCast, _forVotesToCast, _abstainVotesToCast);
+    GOVERNOR.castVoteWithReasonAndParams(
+      proposalId,
+      unusedSupportParam,
+      "rolled-up vote from governance token holders", // Reason string.
```

Comment:
Random thought: we could pull this into a virtual method implemented by the inheritor to allow for each client to implement a custom reason string. If you'd prefer to pull this into an issue and we can discuss the idea later, that's fine.

Reply:
Oh, I like that idea!

Reply:
Done in 9228936, thanks for the suggestion